<launch>

<group ns="ugv3">

    <arg name="altitude" default = "0.0"/>
    <node pkg="uavros_uavugv_sitl" type="leader_estimator_node" name="leader_estimator" output="screen">
        <param name="alt_sp" value = "$(arg altitude)"/>
        <param name="alpha" value = "0.5"/>
        <param name="esm_x_init" value = "0.49"/>
        <param name="esm_y_init" value = "0.58"/>
        <param name="neighbor1_name" value = "virtual_leader_line"/>
        <param name="virtual_leader_vx" value = "0.05"/>
        <param name="virtual_leader_vy" value = "0.05"/>
    </node>

    <node pkg="uavros_uavugv_sitl" type="ackerman_controller_node" name="ackerman_controller" output="screen">
        <param name="v_max" value = "1"/>
        <param name="steering_max_rad" value = "0.2"/>
    	<param name="Kp_x" value = "0.5"/>
        <param name="Kp_y" value = "0.5"/>
        <param name="Kp_heading" value = "1.0"/>
        <param name="init_phi" value = "0"/>
        <param name="wheelbase" value = "1.054"/>
    </node>	

</group>


  <arg name="visualization" default="true"/>
  <group if="$(arg visualization)">
      <node type="rviz" name="rviz" pkg="rviz" args="-d $(find uavros_uavugv_sitl)/config/rviz_uavrover.rviz" />
  </group>

</launch>
